speechbrain/lobes/models/transformer/Conformer.py

```diff
@@ -414,9 +417,9 @@ def forward(
             The mask for the src keys per batch.
         pos_embs: torch.Tensor, torch.nn.Module, optional
             Module or tensor containing the input sequence positional embeddings
-        chunk_size: int
-            Whether to preform convolution chunking to hide future context,
-            useful for chunked conformers in a dynamic chunk training setting
+        dct_config: Optional[DCTConfig]
```

Comment:
What do you mean exactly?

When *actually* streaming, `dct_config` is `None` here (as the streaming function is used instead). This function is specifically the masking-capable code path used during training.

At this specific point in the code, the input shape is identical whether you're in a DCT batch or not (because in the transformer, DCT merely causes masking to prevent attention towards the future).  
When DCT is active for the current batch, this will split up the input in order to apply the DCConv logic. At the end, everything gets concatenated back.

So when training, during a DCT batch, the input and the output of a conformer layer have a strictly identical shape to a non-DCT batch, simply with additional constraints on which frames could attend to which frames.